src/Internal/Service/PaymentRequestService.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Class PaymentRequestService
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Service;
+
+use WC_Payments_API_Payment_Intention;
+use WCPay\Core\Server\Request\Create_And_Confirm_Intention;
+use WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Exceptions\Server\Request\Extend_Request_Exception;
+use WCPay\Core\Exceptions\Server\Request\Immutable_Parameter_Exception;
+use WCPay\Internal\Payment\PaymentContext;
+
+/**
+ * Service for managing requests, connecting payments and the API.
+ */
+class PaymentRequestService {
+	/**
+	 * Creates a payment intent. To be replaced by an abstraction soon.
+	 *
+	 * @param PaymentContext $context Context for the payment.
+	 * @return WC_Payments_API_Payment_Intention
+	 * @throws Invalid_Request_Parameter_Exception
+	 * @throws Extend_Request_Exception
+	 * @throws Immutable_Parameter_Exception
+	 */
+	public function create_intent( PaymentContext $context ) {
+		$request = Create_And_Confirm_Intention::create();
```

Comment:
Nitpick: Perhaps we can create a function inside the request class called :from_payment_context($context) and call these setters if we are planning to use it in multiple places.

Reply:
One of the ideas within `src` is that all static methods must be pure. Your suggestion would be harder to test. However, we must also consider how to construct those requests without the static `create` method, through a dependency instead.

Reply:
Creating a request class via DI, would be a great improvement, IMHO. We can keep it as it is now, and later on update it.